packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @param {long} offset - the starting index of the results the caller wants
+   * @param {long} size - the number of conversations the caller wants
+   * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the specified content summary
+   */
+  @waitForValue('@')
+  getPagedContentSummary(reportId, offset, size) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: `reports/${reportId}/contents/summary`,
+      qs: {offset, size}
+    })
+      .then((res) => res)
+      .catch((reason) => {
+        this.logger.error(reason);
+        return reason;
+      });
+  },
+
+  /**
+   * Generates the report content and triggers the download of the content
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @returns {List<Activities>} HttpResponse
+   */
+  @waitForValue('@')
+  generateReport(reportId) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    let offset = 0;
+    const size = 100;
+    const reportGenerator = new ReportGenerator({reportId});
+
+    function decryptNextActivities(plugin) {
+      return plugin.request({
+        method: 'GET',
+        service: 'ediscovery',
+        resource: `reports/${reportId}/contents`,
+        qs: {offset, size}
+      })
+        .then((res) => {
+          const contents = res.body;
+          return plugin._decryptActivities(contents);
+        })
+        .then((activities) => {
+          if (activities && Array.isArray(activities) && activities.length > 0) {
+            reportGenerator.add(activities);
+            if (activities.length < size) {
+              // No more activities to decrypt
+            }
+            else {
+              // Add activities to the report, increment and decrypt next activities
+              offset += size;
+              decryptNextActivities(plugin);
+            }
+          }
+        })
+        .catch((reason) => {
+          plugin.logger.error(reason);
+          return reason;
+        });
+    }
+
+    return decryptNextActivities(this).then(() => reportGenerator.saveAs());
+  },
+
+  /**
+   * Decrypt the activities found in the report content
+   *
+   * @param {Object[]} activities An array of activities with encrypted messages
+   * @returns {Promise<[any]>} The activities with their messages decrypted
+   */
+  _decryptActivities(activities) {
+    const promises = [];
+    for (let i = 0; i < activities.length; i += 1) {
+      const activity = activities[i];
+      if (activity.verb === 'post' && activity.objectDisplayName && activity.encryptionKeyUrl) {
+        const textPromise = this.spark.internal.encryption.decryptText(activity.encryptionKeyUrl, activity.objectDisplayName, {onBehalfOf: activity.actorId})
```

Comment:
Is there any reason we can't use the existing decrypts for activities?

Reply:
I wasn't aware of an existing decrypt method for activities, is there something available in a plugin aside from encryption?